wgengine/magicsock/magicsock.go

```diff
@@ -338,6 +338,14 @@ type Conn struct {
 
 	closed bool // Close was called
 
+	// interEps (intermediate Endpoints) holds the endpoints that are
+	// discovered in Ping messages received from peers. When a peer is
+	// located in private network, it could potentially see more/different
+	// endpoint addresses of ourselves from STUN. And this extra information
+	// could help other peers in the same private network. See issu #5396
+	// for details.
+	interEps map[netip.AddrPort]bool
```

Comment:
Simply adding all discovered endpoints to a map and never removing them seems problematic. If a host has changing ips and is running for long this would lead to an indefinitely growing list.

This problem could be solved by only keeping the newest endpoint for each remote peer. So basically `map[somepeerid]netip.AddrPort`

Reply:
Thanks for commenting. Changed according to your suggestion.